engine/access/rest/request_decorator.go

```diff
@@ -0,0 +1,145 @@
+	}
+
+	if selectFields, found := middleware.GetFieldsToSelect(r); found {
+		decoratedReq.selectFields = selectFields
+	}
+
+	return decoratedReq
+}
+
+func sliceToMap(values []string) map[string]bool {
+	valueMap := make(map[string]bool, len(values))
+	for _, v := range values {
+		valueMap[v] = true
+	}
+	return valueMap
+}
+
+func (rd *request) expands(field string) bool {
+	return rd.expandFields[field]
+}
+
+func (rd *request) selects() []string {
+	return rd.selectFields
+}
+
+func (rd *request) getVar(name string) string {
+	vars := mux.Vars(rd.Request)
+	return vars[name] // todo(sideninja) consider returning err if non-existing
+}
+
+func (rd *request) getQueryParam(name string) string {
+	return rd.Request.URL.Query().Get(name) // todo(sideninja) consider returning err if non-existing
+}
+
+func (rd *request) getQueryParams(name string) []string {
+	param := rd.Request.URL.Query().Get(name)
+	// currently, the swagger generated Go REST client is incorrectly doing a `fmt.Sprintf("%v", id)` for the id slice
+	// resulting in the client sending the ids in the format [id1 id2 id3...]. This is a temporary workaround to
+	// accommodate the client for now by doing a strings.Fields if commas are not present.
+	// Issue to to fix the client: https://github.com/onflow/flow/issues/698
+	param = strings.TrimSuffix(param, "]")
+	param = strings.TrimPrefix(param, "[")
+	if len(param) == 0 {
+		return nil
+	}
+	if strings.Contains(param, ",") {
+		return strings.Split(param, ",")
+	}
+	return strings.Fields(param)
+}
+
+func (rd *request) bodyAs(dst interface{}) error {
+	//todo(sideninja) validate size
+
+	dec := json.NewDecoder(rd.Body)
+	dec.DisallowUnknownFields()
+
+	err := dec.Decode(&dst)
+	if err != nil {
+		var syntaxError *json.SyntaxError
+		var unmarshalTypeError *json.UnmarshalTypeError
+
+		switch {
+		case errors.As(err, &syntaxError):
+			err := fmt.Errorf("request body contains badly-formed JSON (at position %d)", syntaxError.Offset)
+			return NewBadRequestError(err)
+
+		case errors.Is(err, io.ErrUnexpectedEOF):
+			err := fmt.Errorf("request body contains badly-formed JSON")
+			return NewBadRequestError(err)
+
+		case errors.As(err, &unmarshalTypeError):
+			err := fmt.Errorf("request body contains an invalid value for the %q field (at position %d)", unmarshalTypeError.Field, unmarshalTypeError.Offset)
+			return NewBadRequestError(err)
+
+		case strings.HasPrefix(err.Error(), "json: unknown field "):
+			fieldName := strings.TrimPrefix(err.Error(), "json: unknown field ")
+			err := fmt.Errorf("Request body contains unknown field %s", fieldName)
+			return NewBadRequestError(err)
+
+		case errors.Is(err, io.EOF):
+			err := fmt.Errorf("request body must not be empty")
+			return NewBadRequestError(err)
+
+		case err.Error() == "http: request body too large":
+			err := fmt.Errorf("request body must not be larger than 1MB")
+			return NewRestError(http.StatusRequestEntityTooLarge, err.Error(), err)
+
```

Comment:
I'm a bit confused about this part. How is it possible for the error returned from a json Decode to be an HTTP error?

Reply:
You are correct, this is not relevant, an oversight will fix.